client/blocks/credit-card/use-payment-processing.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import {
+	CardElement,
+	CardNumberElement,
+	useElements,
+} from '@stripe/react-stripe-js';
+
+/**
+ * Internal dependencies
+ */
+import { PAYMENT_METHOD_NAME } from './constants';
+import {
+	getStripeServerData,
+	getErrorMessageForTypeAndCode,
+} from '../stripe-utils';
+import { errorTypes } from '../stripe-utils/constants';
+
+/**
+ * @typedef {import('@stripe/stripe-js').Stripe} Stripe
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').EventRegistrationProps} EventRegistrationProps
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').BillingDataProps} BillingDataProps
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').EmitResponseProps} EmitResponseProps
+ * @typedef {import('react').Dispatch<string>} SourceIdDispatch
+ */
+
+/**
+ * @typedef {function(function():any):function():void} EventRegistration
+ */
+
+/**
+ * A custom hook that registers stripe payment processing with the
+ * onPaymentProcessing event from checkout.
+ *
+ * @param {function(any):string} onStripeError       Sets an error for stripe.
+ * @param {string}               error               Any set error message (an empty string if no
+ *                                                   error).
+ * @param {Stripe}               stripe              The stripe utility
+ * @param {BillingDataProps}     billing             Various billing data items.
+ * @param {EmitResponseProps}    emitResponse        Various helpers for usage with observer
+ *                                                   response objects.
+ * @param {string}               sourceId            Current set stripe source id.
+ * @param {SourceIdDispatch}     setSourceId         Setter for stripe source id.
+ * @param {EventRegistration}    onPaymentProcessing The event emitter for processing payment.
+ */
+export const usePaymentProcessing = (
+	onStripeError,
+	error,
+	stripe,
+	billing,
+	emitResponse,
+	sourceId,
+	setSourceId,
+	onPaymentProcessing
+) => {
+	const elements = useElements();
+	// hook into and register callbacks for events
+	useEffect( () => {
+		const createSource = async ( ownerInfo ) => {
+			const elementToGet = getStripeServerData().inline_cc_form
+				? CardElement
+				: CardNumberElement;
+			return await stripe.createSource(
+				// @ts-ignore
+				elements?.getElement( elementToGet ),
+				{
+					type: 'card',
+					owner: ownerInfo,
+				}
+			);
+		};
+		const onSubmit = async () => {
+			try {
+				const billingData = billing.billingData;
+				// if there's an error return that.
+				if ( error ) {
+					return {
+						type: emitResponse.responseTypes.ERROR,
+						message: error,
+					};
+				}
+				// use token if it's set.
+				if ( sourceId !== '' && sourceId !== '0' ) {
```

Comment:
Let's use the established [yoda condition WP standard](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions) even though that's _technically_ only defined for PHP 🙂

Reply:
This seems like a larger question that all Woo teams should align on (see my earlier comment regarding how Woo eslint-plugin is currently configured). I thought that decision had already been made, but maybe it hasn't.

Reply:
I'm not aware of a decision like that, but truth be told I haven't looked 🤔 .

It's definitely something to think about! Thanks for bringing it up 🙏